### PR TITLE
LA-182 Copy To My Space From Shared Space

### PR DESCRIPTION
**Pull request by imGok**
copycopy

### CODE REVIEW

File: domain/lib/src/usecases/myspace/copy_to_my_space_interactor.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import 'package:domain/src/state/success.dart';
+import 'package:domain/src/usecases/myspace/my_space_view_state.dart';
+
+class CopyToMySpaceInteractor {
+  final DocumentRepository _documentRepository;
+
+  CopyToMySpaceInteractor(this._documentRepository);
+
+  Future<Either<Failure, Success>> execute(CopyRequest copyRequest) async {
+    final resultState = await catching(() => _documentRepository.copyToMySpace(copyRequest))
```
Comment (manhduc-nguyen) [inline domain/lib/src/usecases/myspace/copy_to_my_space_interactor.dart:46]: I think you should use `try catch` for this. It is much easier to read.

File: lib/presentation/localizations/app_localizations.dart [hunk 2]
```diff
@@ -631,6 +631,27 @@ class AppLocalizations {
       args: [nbFiles, quota, firstElementName],
     );
   }
+
+  String get copy_to_my_space {
```
Comment (manhduc-nguyen) [inline lib/presentation/localizations/app_localizations.dart:635]: pls also add translated string (or even untranslated string) to `intl_fr.arb` and other localization files.

File: domain/lib/src/repository/document/document_repository.dart [hunk 3]
```diff
@@ -54,4 +55,6 @@ abstract class DocumentRepository {
   Future<Uri> downloadDocumentIOS(Document document, Token token, Uri baseUrl, CancelToken cancelToken);
 
   Future<Document> remove(DocumentId documentId);
+
+  Future<List<Document>> copyToMySpace(CopyRequest copyRequest);
```
Comment (hoangdat) [inline domain/lib/src/repository/document/document_repository.dart:59]: why copy 1 file, but the return is List of document?

File: domain/lib/src/usecases/myspace/copy_to_my_space_interactor.dart [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+//
+
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:domain/src/repository/document/document_repository.dart';
+import 'package:domain/src/state/failure.dart';
+import 'package:domain/src/state/success.dart';
+import 'package:domain/src/usecases/myspace/my_space_view_state.dart';
+
+class CopyToMySpaceInteractor {
```
Comment (hoangdat) [inline domain/lib/src/usecases/myspace/copy_to_my_space_interactor.dart:40]: How about copy multiple file to MySpace?

File: domain/lib/src/repository/document/document_repository.dart (hunk 3 above)
Comment (imGok) [reply domain/lib/src/repository/document/document_repository.dart:59]: My bad, I took android version for example but yes it seems the API returns only one doc (cf API doc)

File: domain/lib/src/usecases/myspace/copy_to_my_space_interactor.dart (hunk 4 above)
Comment (imGok) [reply domain/lib/src/usecases/myspace/copy_to_my_space_interactor.dart:40]: On the way

File: domain/lib/src/repository/document/document_repository.dart (hunk 3 above)
Comment (imGok) [reply domain/lib/src/repository/document/document_repository.dart:59]: And my bad again after investigation API returns a list !

**Comment by imGok** [comment]
@manhduc-nguyen @hoangdat check again pls :))

File: lib/l10n/intl_messages.arb [hunk 5]
```diff
@@ -482,5 +482,20 @@
       "quota": {},
       "firstElementName": {}
     }
+  },
+  "copy_to_my_space": "Copy to My Space",
```
Comment (manhduc-nguyen) [inline lib/l10n/intl_messages.arb:486]: we are missing `notEnoughQuota` and `tooBigFiles` in this file?

**Review by manhduc-nguyen — APPROVED** [review]
LGTM!

**Review by hoangdat — APPROVED** [review]

**Pull request merged**